Patches/PlayerControlPatch.cs

```diff
@@ -1620,18 +1624,21 @@ public static bool Prefix(PlayerControl __instance, [HarmonyArgument(0)] ref Rol
                 if (target.IsAnySubRole(x => x.IsGhostRole()) || target.GetCustomRole().IsGhostRole())
                 {
                     roleType = RoleTypes.GuardianAngel;
+                    return true;
                 }
                 else if (ghostRoles.All(kvp => kvp.Value == RoleTypes.CrewmateGhost))
                 {
                     roleType = RoleTypes.CrewmateGhost;
+                    return true;
                 }
                 else if (ghostRoles.All(kvp => kvp.Value == RoleTypes.ImpostorGhost))
                 {
                     roleType = RoleTypes.ImpostorGhost;
+                    return true;
                 }
                 else
                 {
-                    foreach ((var seer, var role) in ghostRoles)
+                    foreach ((var seer, var role) in ghostRoles) // Idk wtf this for tbh
                     {
                         Logger.Info($"Desync {targetName} => {role} for {seer.GetNameWithRole().RemoveHtmlTags()}", "PlayerControl.RpcSetRole");
                         target.RpcSetRoleDesync(role, seer.GetClientId());
```

Comment:
This code assigns the roles of ghosts desync, that is, if the role is non-impostor, this player will assign ImpostorGhost, but for others players it's will be assigned as CrewmateGhost, if you test this in the game and look at the logs, then everything will become clear